gallery/src/PageHeader/HostedVersionsSelect.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, ChangeEventHandler, useState } from 'react';
+import { NxFormSelect } from '@sonatype/react-shared-components';
+
+import packageJson from '../../package.json';
+
+import './HostedVersionsSelect.scss';
+
+const CURRENT_VERSION = packageJson.version;
+const HOSTED_VERSIONS_JSON_URL = 'https://gallery.sonatype.dev/versions/latest/hosted-versions.json';
+
+const getHostedVersion = () => fetch(HOSTED_VERSIONS_JSON_URL, {
+  headers: {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json'
+  }
+}).then((response) => {
+  if (!response.ok) {
+    throw new Error('Failed to load hosted versions.');
+  }
+  return response.json();
+});
+
+const HostedVersionsSelect = () => {
+  const [isLoadingHostedVersions, setIsLoadingHostedVersions] = useState(false);
+  const [hostedVersions, setHostedVersions] = useState([CURRENT_VERSION]);
+
+  const onChange: ChangeEventHandler<HTMLSelectElement> = event =>
+    window.location.href = `/versions/${event.target.value}`;
+
+  useEffect(() => {
+    setIsLoadingHostedVersions(true);
+    getHostedVersion()
+        .then((hostedVersions) => {
+          setHostedVersions(hostedVersions);
+          setIsLoadingHostedVersions(false);
+        })
+        .catch(() => {
+          setHostedVersions([CURRENT_VERSION]);
+          setIsLoadingHostedVersions(false);
+        });
+  }, []);
+
+  return (
+    <div className="hosted-versions-select">
```

Comment:
Classnames here should start with `gallery-`

Reply:
> Classnames here should start with `gallery-`

Fixed here: d3041d6fa05deb382ed71116327f25d8b5e5ffc6